src/confluent_kafka/src/Admin.c

```diff
@@ -1531,6 +1528,361 @@ const char Admin_list_consumer_groups_doc[] = PyDoc_STR(
+                                goto err;
+                        }
+                        cfl_PyObject_GetInt(mechanism,"value",&mechanism_val,0,1);
+
+                        c_alterations[i] = rd_kafka_UserScramCredentialDeletion_new(user_c,(rd_kafka_ScramMechanism_t)mechanism_val);
+                        Py_DECREF(mechanism);
+                        mechanism = NULL;
+                }
+
+                Py_DECREF(u_user);
+                Py_XDECREF(uo_user);
+        }
+        /* Use librdkafka's background thread queue to automatically dispatch
+         * Admin_background_event_cb() when the admin operation is finished. */
+        rkqu = rd_kafka_queue_get_background(self->rk);
+
+        /*
+         * Call AlterConfigs
+         *
+         * We need to set up a CallState and release GIL here since
+         * the event_cb may be triggered immediately.
+         */
+        CallState_begin(self, &cs);
+        rd_kafka_AlterUserScramCredentials(self->rk, c_alterations, alteration_cnt, c_options, rkqu);
+        CallState_end(self, &cs);
+
+        if(c_alterations) {
+                for(i=0;i<alteration_cnt;i++)
+                        rd_kafka_UserScramCredentialAlteration_destroy(c_alterations[i]);
+                free(c_alterations);
+        }
+        rd_kafka_queue_destroy(rkqu); /* drop reference from get_background */
+        rd_kafka_AdminOptions_destroy(c_options);
+        Py_DECREF(UserScramCredentialAlteration_type); /* from lookup() */
+        Py_DECREF(UserScramCredentialUpsertion_type); /* from lookup() */
+        Py_DECREF(UserScramCredentialDeletion_type); /* from lookup() */
+        Py_DECREF(ScramCredentialInfo_type); /* from lookup() */
+        Py_DECREF(ScramMechanism_type); /* from lookup() */
+        Py_RETURN_NONE;
+err:
+
+        Py_XDECREF(u_user);
+        Py_XDECREF(uo_user);
+        Py_XDECREF(salt);
+        Py_XDECREF(password);
+        Py_XDECREF(scram_credential_info);
+        Py_XDECREF(mechanism);
+
+        Py_DECREF(UserScramCredentialAlteration_type); /* from lookup() */
+        Py_DECREF(UserScramCredentialUpsertion_type); /* from lookup() */
+        Py_DECREF(UserScramCredentialDeletion_type); /* from lookup() */
+        Py_DECREF(ScramCredentialInfo_type); /* from lookup() */
+        Py_DECREF(ScramMechanism_type); /* from lookup() */
+
+        if(c_alterations) {
+                for(i=0;i<alteration_cnt;i++)
+                        rd_kafka_UserScramCredentialAlteration_destroy(c_alterations[i]);
+                free(c_alterations);
+        }
+        if (c_options) {
+                rd_kafka_AdminOptions_destroy(c_options);
+                Py_DECREF(future);
+        }
+        return NULL;
+}
+
+
+const char alter_user_scram_credentials_doc[] = PyDoc_STR(
+        ".. py:function:: alter_user_scram_credentials(alterations)\n"
+        "\n"
+        "  Alters the credentials for a user.\n"
+        "  Supported : Upsertion , Deletion.\n");
```

Comment:
1. Add parameters future and request_timeout
2. Add note not to use this function directly.